versions/development/SPEC.md

```diff
@@ -760,7 +760,9 @@ Engines should at the very least support the following protocols for import URIs
 
 * `http://` and `https://`
 * `file://`
-* no protocol (which should be interpreted as `file://`
+* No protocol (see below)
+
+In the event that there is no protocol, if this is an absolute path (i.e. starts with `/`) it is interpreted as a `file://`. Otherwise it is **relative** to the location of the current document. For instance, if the current doc is at `file://foo/bar/baz/qux.wdl` and specificies `import some/task.wdl` this will be resolved to `file://foo/bar/baz/some/task.wdl`. Likewise if the current doc is at `http://www.github.com/openwdl/coolwdls/myWorkflow.wdl` and imports `subworkflow.wdl`, it will be resolved to `http://www.github.com/openwdl/coolwdls/subworkflow.wdl`. It is up to the implementation to provide a mechanism which will allow these imports to be resolved correctly.
```

Comment:
I've got to believe there's a more sensible `/` for HTTP?

eg if I were to say `import /chrisl/coolwdl/task.wdl` from `http://www.github.com/openwdl/coolwdls/myWorkflow.wdl` I might expect that to look for `http://www.github.com/chrisl/coolwdl/task.wdl` rather than on the local FS

Reply:
I thought about that a bit. My objection is that in file system semantics `/xyz` implies an absolute path. Yes, I did just complain about people assuming filesystem semantics above, but this feels like a case where more people will be bit by it than helped. 

OTOH I do like the simplicity of what you describe.

Reply:
@cjllanwarne although what you describe is simple it seems like you could quickly run into a frustrating situation where people have a hodge podge of imports and nested calls that cannot be reasily imported. For example if a restapi exposes a wdl at:
```http://www.wdl-host.com/public/my-workflow.wdl```
and that wdl imports something from `/....`, it becomes a bit ambigious as to what the `/` relative path means. when the workflow was written and then exposed by a restapi it may not have known anything about where it was going to be  hosted. Or even if it was going to be hosted at all.

Personally I do not really like any paths that start with `/` but really everything should be relative to the current WDL (other then http/ftp`) calls. The moment we start allowing absolute paths everything becomes alot harder and is more subject to breaking because of permissions / moved or deleted file